light/provider/http/http.go

```diff
@@ -113,11 +118,12 @@ func (p *http) validatorSet(ctx context.Context, height *int64) (*types.Validato
 			if err != nil {
 				// TODO: standardize errors on the RPC side
 				if regexpMissingHeight.MatchString(err.Error()) {
-					return nil, provider.ErrLightBlockNotFound
+					return nil, p.noBlock()
 				}
 				// if we have exceeded retry attempts then return no response error
 				if attempt == maxRetryAttempts {
-					return nil, provider.ErrNoResponse
+					p.noResponseCount++
```

Comment:
why are u incrementing `noResponseCount ` both here and in `noResponse`?

Reply:
Ah yeah good catch